.github/workflows/phpunit.yml

```diff
@@ -0,0 +1,99 @@
+name: PHPUnit tests
+
+on: pull_request
+
+env:
+  WP_TRAVISCI: "phpunit"
+  COMPOSER_ROOT_VERSION: "dev-master"
+
+jobs:
+  phpunit:
+    name: "PHP: ${{ matrix.php }} WP: ${{ matrix.wp }}"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        php: ["5.6", "7.0", "7.2", "7.3", "7.4snapshot"]
+        wp: ["master", "latest", "previous"]
+        exclude:
+        # excludes node 4 on macOS
+        - php: "7.0"
+        - php: "7.2"
+        - wp: "latest"
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          tools: composer, phpunit:v7
+          extensions: mysql, imagick
+          coverage: none
+
+      - name: Get Composer cache directory
+        id: composer-cache
+        run: |
+          echo "::set-output name=dir::$(composer config cache-files-dir)"
+      - name: Composer cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+
+      - name: Tool versions
+        run: |
+          php --version
+          composer --version
+
+      - name: Setup and run tests
+        env:
+          WP_BRANCH: ${{ matrix.wp }}
+          TRAVIS_PHP_VERSION: ${{ matrix.php }}
+        run: |
+          sudo systemctl start mysql.service
+          export PLUGIN_SLUG=$(basename $(pwd))
+          export PATH="$HOME/.config/composer/vendor/bin:$HOME/.composer/vendor/bin:$PATH"
+          ./tests/setup-travis.sh
+          ./tests/run-travis.sh
```

Comment:
It's up to you, but I'd recommend moving all the stuff from setup-travis.sh and run-travis.sh in here into this file, or into .github/actions/ that are then called from here.

run-travis.sh, in particular, seems kind of spaghetti-ish and seems like it would be more clear split up into steps.

If you don't want to, that's fine with me but I may well do it in a followup :wink:

Reply:
yeah, totally! I'll look at what can be done

Reply:
I guess I'm going to refactor it in a follow-up PR.